src/__experimental__/components/date/date.spec.js

```diff
@@ -480,6 +480,16 @@ describe('Date', () => {
       });
     });
 
+    it('short circuits when either "disabled" and/or "readOnly"', () => {
```

Comment:
I think the correct implementation of that test would be to have separate `it` block for every permutation of these props.
1. disabled
2. readOnly
3. disabled and readOnly
4. neither (probably optional)